plugin/pkg/admission/broker/authsarcheck/admission.go

```diff
@@ -71,57 +71,111 @@ func (s *sarcheck) Admit(a admission.Attributes) error {
+			secretRef = clusterServiceBroker.Spec.AuthInfo.Basic.SecretRef
+		} else if clusterServiceBroker.Spec.AuthInfo.Bearer != nil {
+			secretRef = clusterServiceBroker.Spec.AuthInfo.Bearer.SecretRef
+		}
+
+		if secretRef == nil {
+			return nil
+		}
+		klog.V(5).Infof("ClusterServiceBroker %+v: evaluating auth secret ref, with authInfo %q", clusterServiceBroker, secretRef)
+		userInfo := a.GetUserInfo()
+
+		sar := &authorizationapi.SubjectAccessReview{
+			Spec: authorizationapi.SubjectAccessReviewSpec{
+				ResourceAttributes: &authorizationapi.ResourceAttributes{
+					Namespace: secretRef.Namespace,
+					Verb:      "get",
+					Group:     corev1.SchemeGroupVersion.Group,
+					Version:   corev1.SchemeGroupVersion.Version,
+					Resource:  corev1.ResourceSecrets.String(),
+					Name:      secretRef.Name,
+				},
+				User:   userInfo.GetName(),
+				Groups: userInfo.GetGroups(),
+				Extra:  convertToSARExtra(userInfo.GetExtra()),
+				UID:    userInfo.GetUID(),
+			},
+		}
+		sar, err := s.client.AuthorizationV1().SubjectAccessReviews().Create(sar)
+		if err != nil {
+			return err
+		}
+
+		if !sar.Status.Allowed {
+			return admission.NewForbidden(a, fmt.Errorf("broker forbidden access to auth secret (%s): Reason: %s, EvaluationError: %s", secretRef.Name, sar.Status.Reason, sar.Status.EvaluationError))
+		}
 		return nil
-	}
-	klog.V(5).Infof("ClusterServiceBroker %+v: evaluating auth secret ref, with authInfo %q", clusterServiceBroker, secretRef)
-	userInfo := a.GetUserInfo()
-
-	sar := &authorizationapi.SubjectAccessReview{
-		Spec: authorizationapi.SubjectAccessReviewSpec{
-			ResourceAttributes: &authorizationapi.ResourceAttributes{
-				Namespace: secretRef.Namespace,
-				Verb:      "get",
-				Group:     corev1.SchemeGroupVersion.Group,
-				Version:   corev1.SchemeGroupVersion.Version,
-				Resource:  corev1.ResourceSecrets.String(),
-				Name:      secretRef.Name,
+	} else if a.GetResource().GroupResource() == servicecatalog.Resource("servicebrokers") {
+		serviceBroker, ok := a.GetObject().(*servicecatalog.ServiceBroker)
+		if !ok {
+			return errors.NewBadRequest("Resource was marked with kind ServiceBroker, but was unable to be converted")
+		}
+
+		if serviceBroker.Spec.AuthInfo == nil {
+			// no auth secret to check
+			return nil
+		}
+
+		var secretRef *servicecatalog.LocalObjectReference
+		if serviceBroker.Spec.AuthInfo.Basic != nil {
+			secretRef = serviceBroker.Spec.AuthInfo.Basic.SecretRef
+		} else if serviceBroker.Spec.AuthInfo.Bearer != nil {
+			secretRef = serviceBroker.Spec.AuthInfo.Bearer.SecretRef
+		}
+
+		if secretRef == nil {
+			return nil
+		}
+		klog.V(5).Infof("ServiceBroker %+v: evaluating auth secret ref, with authInfo %q", serviceBroker, secretRef)
+		userInfo := a.GetUserInfo()
+
+		sar := &authorizationapi.SubjectAccessReview{
+			Spec: authorizationapi.SubjectAccessReviewSpec{
+				ResourceAttributes: &authorizationapi.ResourceAttributes{
+					Namespace: serviceBroker.Namespace,
```

Comment:
this seems to be the only servicebroker specific thing. can we save it off into a namespace variable in the earlier part?